pkg/test-fw/tests/trace_open_test.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2019-2021 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests
+
+import (
+	"io/fs"
+	"testing"
+
+	"github.com/inspektor-gadget/inspektor-gadget/integration"
+	IG "github.com/inspektor-gadget/inspektor-gadget/pkg/test-fw/ig"
+	eventtypes "github.com/inspektor-gadget/inspektor-gadget/pkg/types"
+)
+
+type EventType string
+
+const ImageName = "docker.io/pawarpranav83/trace_open:latest"
+
+type Event struct {
+	eventtypes.CommonData
+
+	// Type indicates the kind of this event
+	Type EventType `json:"type"`
+
+	// Message when Type is ERR, WARN, DEBUG or INFO
+	Message string `json:"message,omitempty"`
+}
+
+// A user would have to implement their own event type as well, therefore not using this from ig repo
+type traceOpenEvent struct {
+	Event
+	eventtypes.WithMountNsID
```

Comment:
Image based gadgets don't base their events on these types. I think it's better to define the mount namespace, (and other interesting fields directly on this structure. As done for other fields below.